templates/CRM/Admin/Form/FieldIsOriginalEmlAttached.tpl

```diff
@@ -0,0 +1,23 @@
+<table style="display: none">
+  <tr class="crm-mail-settings-form-block-is_original_eml_attached"><td class="label">&nbsp;</td><td>{$form.is_original_eml_attached.html}{$form.is_original_eml_attached.label} {help id='is_original_eml_attached'}</td></tr>
+</table>
+
+{literal}
+<script type="text/javascript">
+(function($) {
+  var $form = $('form.{/literal}{$form.formClass}{literal}');
+
+  function showActivityFields() {
+    var fields = [
+      '.crm-mail-settings-form-block-is_original_eml_attached',
+    ];
+    $(fields.join(', '), $form).toggle($(this).val() === '0');
+  }
+
+  // Add new field to the form.
+  CRM.$('.crm-mail-settings-form-block-is_original_eml_attached').insertBefore(CRM.$('.crm-mail-settings-form-block-activity_status'));
+  // Show/hide field when 'Used For?' selected value is changed.
+  $('select[name="is_default"]').each(showActivityFields).change(showActivityFields);
+})(CRM.$);
```

Comment:
```js
(function ($) {
  var $form = $('form.{/literal}{$form.formClass}{literal}');

  (function init () {
    // Add new field to the form.
    $('.crm-mail-settings-form-block-is_original_eml_attached')
      .insertBefore('.crm-mail-settings-form-block-activity_status'));

    $('select[name="is_default"]')
      .each(showActivityFields)
      .change(showActivityFields);
  })();

  // Show/hide field when 'Used For?' selected value is changed.
  function showActivityFields() {
    var fields = [
      '.crm-mail-settings-form-block-is_original_eml_attached',
    ];
    $(fields.join(', '), $form).toggle($(this).val() === '0');
  }
})(CRM.$);
```

Reply:
We use the init function to define the basic logic the form has. We should try to make it as human readable as possible so in theory we should have something like this in the init function:

```js
(function init () {
  moveEmlFieldBeforeActivityStatus();
  handleEmlFieldChanges();
}();
```

but since this is a local script you don't have to go too in-depth